File: generate_random_points.m
function [points] = generate_random_points(x,y,z,zmin,n)
    %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
    %%% Generador de n puntos uniformemente distribuidos      %%%
    %%% dentro de un espacio poliedrico centrado en el orígen %%%
    %%% de dimensiones dadas.                                 %%%
    %%% x,y,z son las dimensiones del espacio y zmin es la    %%%
    %%% distancia al centro de coordenadas.                   %%%
    %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
    rng('default')
    dims = [x y z];
    coords = zeros(2, length(dims));
    for i=2:length(dims)
        coords(1,i) = -dims(i)/2;
        coords(2,i) = dims(i)/2;
    end
    coords(:,1) = [-1 x+1]; 
    coords(:,3) = coords(:,3)+ (zmin + z/2);
    for i=1:length(coords)
        points(:,i) = coords(1,i) + (coords(2,i) - coords(1,i))*rand(n,1);
    end

end 

File: generate_random_points.py
import numpy as np

def generate_random_points(x,y,z,zmin,n):

    dims = [x, y, z]

    coords = np.zeros((2, 3))
    points = np.zeros((n, 3))

    coords[:,0] = np.array([-0.8, x + 0.8])
    coords[:,1] = np.array([-y/2, y/2])
    coords[:,2] = np.array([zmin, zmin+z])

    for i in range(3):

        points[:,i] = coords[0,i] + (coords[1,i] - coords[0,i])*np.random.uniform(0, 1, n)
    
    return points


if __name__ == "__main__":
    a = generate_random_points(5, 2, 6, 10, 1)
    print(a)